app/logic/events.py

```diff
@@ -211,7 +211,21 @@ def getUpcomingEventsForUser(user, asOf=datetime.datetime.now()):
                     .order_by(Event.startDate, Event.name).execute() # keeps the order of events the same when the dates are the same
                     )
 
-    return events
+
+    events_list = []
+    curr_recurr_id = None
+
+    # removes all recurring events except for the next upcoming one
+    for event in events:
+        if event.recurringId:
+            if event.recurringId != curr_recurr_id:
+                curr_recurr_id = event.recurringId
+                events_list.append(event)
+
+        else:
+            events_list.append(event)
```

Comment:
This is good but it does not account for multiple recurring events. Currently, you are only checking the recurring event id and making sure it isn't the same as the last recurring id encountered, it would be good to store the recurring ids somewhere so we know if we have encountered that recurring event before.